src/common.speech/ConversationTranslator/ConversationTranslatorRecognizer.ts

```diff
@@ -104,27 +91,118 @@ export class ConversationTranslatorRecognizer extends Recognizer implements ICon
         return this.privProperties;
     }
 
+    public isDisposed(): boolean {
+        return this.privIsDisposed;
+    }
+
+    /**
+     * Connect to the recognizer
+     * @param token
+     */
+    public connect(token: string, cb?: () => void, err?: (e: string) => void): void {
+        try {
+            Contracts.throwIfDisposed(this.privIsDisposed);
+            Contracts.throwIfNullOrWhitespace(token, "token");
+            this.privReco.conversationTranslatorToken = token;
+            this.privReco.connect();
```

Comment:
This is a blocking call. Does this wait for the participant list to be received? If so this is potentially problematic as we don't want to lock up the UI for what could possibly be seconds.

As context, the SpeechRecognizer is designed in a way where you don't explicitly call Connect() or Disconnect(). Instead end users call one of the StartRecognition...Async() methods and internally it handles connecting/disconnecting. Those synchronous methods are exposed for use through the Connection class.

Since you are extending the Recognizer base as is, this is potentially problematic since we do need an explicit asynchronous way to start the connection. Perhaps adding a protected method to the ServiceRecognizerBase class that gives you access to the connecitonImpl promise is what you need to call here

Reply:
This was based on the DialogService implementation.

Reply:
Updated to use Ralph's Async method addition.